src/Nethermind/Nethermind.JsonRpc/Data/AccessListItemForRpc.cs

```diff
@@ -5,45 +5,50 @@
 using System.Collections.Generic;
 using System.Linq;
 using Nethermind.Core;
+using Nethermind.Core.Collections;
 using Nethermind.Core.Eip2930;
 using Nethermind.Int256;
 using Nethermind.Serialization.Json;
 using Newtonsoft.Json;
 
 namespace Nethermind.JsonRpc.Data
 {
-    public class AccessListItemForRpc
+    public struct AccessListItemForRpc : IEquatable<AccessListItemForRpc>
     {
-        public AccessListItemForRpc(Address address, IReadOnlyCollection<UInt256>? storageKeys)
+        public AccessListItemForRpc(Address address, IEnumerable<UInt256>? storageKeys)
         {
             Address = address;
-            StorageKeys = storageKeys?.ToArray() ?? Array.Empty<UInt256>();
+            StorageKeys = storageKeys;
```

Comment:
Should we do `Enumerable.Empty` ?

Reply:
We can do that to avoid `StorageKeys` be nullable.

Reply:
We can't do that since it breaks JSON serialization: we would need to handle nulls as `Enumerable.Empty` at the JSON converter level too. I would suggest to leave it as it is.